news-bundle/src/EventListener/SitemapListener.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\EventListener;
+
+use Contao\Controller;
+use Contao\CoreBundle\Event\ContaoCoreEvents;
+use Contao\CoreBundle\Event\SitemapEvent;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\News;
+use Contao\NewsArchiveModel;
+use Contao\NewsModel;
+use Contao\PageModel;
+use Terminal42\ServiceAnnotationBundle\Annotation\ServiceTag;
+
+/**
+ * @ServiceTag("kernel.event_listener", event=ContaoCoreEvents::SITEMAP)
+ *
+ * @internal
+ */
+class SitemapListener
+{
+    private ContaoFramework $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function __invoke(SitemapEvent $sitemapEvent): void
+    {
+        $sitemap = $sitemapEvent->getDocument();
+        $urlSet = $sitemap->firstChild;
+
+        $this->framework->initialize();
+
+        /** @var NewsArchiveModel $newsArchiveModel */
+        $newsArchiveModelAdapter = $this->framework->getAdapter(NewsArchiveModel::class);
+
+        // Get all news archives
+        $archives = $newsArchiveModelAdapter->findAll();
+
+        if (null === $archives) {
+            return;
+        }
+
+        /** @var PageModel $pageModelAdapter */
```

Comment:
No need to declare the types, this should be automatically inferred in our latest return type comments.

Reply:
True, it's just my IDE that's unable to parse that.